plugins/module_utils/network/sonic/config/dhcp_snooping/dhcp_snooping.py

```diff
@@ -0,0 +1,636 @@
+            # afis parameter only stores the on device config at this point
+            commands = have
+            requests = self.get_delete_all_have_requests(afis)
+        else:
+            # some mix of settings specified in both
+            commands, requests = self.get_delete_specific_requests(afis)
+
+        if commands and len(requests) > 0:
+            commands = update_states(commands, "deleted")
+        else:
+            commands = []
+        return commands, requests
+
+    def _state_overridden(self, want, have, afis):
+        """ The command generator when state is overridden
+
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        commands = []
+        requests = []
+        if not want:
+            return commands, requests
+
+        # Determine if there is any configuration specified in the playbook
+        # that is not contained in the current configuration.
+        diff_requested = get_diff(want, have, TEST_KEYS)
+
+        # Determine if there is anything already configured that is not
+        # specified in the playbook.
+        diff_unwanted = get_diff(have, want, TEST_KEYS)
+
+        # Idempotency check: If the configuration already matches the
+        # requested configuration with no extra attributes, no
+        # commands should be executed on the device.
+        if not diff_requested and not diff_unwanted:
+            return commands, requests
+
+        # Delete all current DHCP snooping configuration
+        commands = [have]
+        requests = self.get_delete_all_have_requests(afis)
+        if commands and len(requests) > 0:
+            commands = update_states(commands, "deleted")
+
+        # Apply the commands from the playbook
+        merged_commands = want
+        overridden_requests = self.get_modify_requests(merged_commands)
+        requests.extend(overridden_requests)
+        if merged_commands and len(overridden_requests) > 0:
+            merged_commands = update_states(merged_commands, "overridden")
+            commands.extend(merged_commands)
+        return commands, requests
+
+    def _state_replaced(self, want, have, afis):
+        """ The command generator when state is replaced
+
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        commands = []
+        requests = []
+
+        # Delete replaced groupings
+        commands, requests = self.get_delete_replaced_groupings(afis)
+        if commands and len(requests) > 0:
+            commands = update_states(commands, "deleted")
+
+        if requests:
+            modify_have = []
+        else:
+            modify_have = have
+
+        # Apply the commands from the playbook
+        diff = get_diff(want, modify_have, TEST_KEYS)
+        merged_commands = diff
```

Comment:
`get_delete_replcaed_groupings` is supposed to only return the commands that are deleted. Made that function only handle that half of the replaced process. 
For the grabbing diff part, I tried changing it, but running into issues and I think it might take a day or two to get it reworked.  For a replaceable subsection like source_bindings, I don't think directly using `get_diff(want, have, TEST_KEYS)` without checking requests would work. Delete part deletes the whole source binding, and then if using get_diff with have, it might not find all settings that should be put in. have still stores the settings before deletion and if a single source binding partially patches in want and have, the replaced call only adds in the differences which is just part of what it should be set to. the check does make all the want requests to be added, but it also makes sure that all of each wanted source binding is added